packages/cli/lib/services/compile.service.ts

```diff
@@ -71,7 +71,17 @@ class CompileService {
 
         for (const file of integrationFiles) {
             try {
-                const providerConfiguration = config.find((config) => [...config.syncs, ...config.actions].find((sync) => sync.name === file.baseName));
+                const isNested =
+                    file.inputPath.split('/')[file.inputPath.split('/').length - 2] === 'syncs' ||
+                    file.inputPath.split('/')[file.inputPath.split('/').length - 2] === 'actions';
+
+                let providerConfiguration;
+                if (isNested) {
+                    const providerConfigKey = file.inputPath.split('/')[file.inputPath.split('/').length - 3];
+                    providerConfiguration = config.find((config) => config.providerConfigKey === providerConfigKey);
+                } else {
+                    providerConfiguration = config.find((config) => [...config.syncs, ...config.actions].find((sync) => sync.name === file.baseName));
+                }
```

Comment:
would storing the result of `file.inputPath.split('/')` and avoid the repetition make the code more readable? ex:

```
const pathSegments = file.inputPath.split('/');
const typeFolder = pathSegments[pathSegments.length - 2];
const isNested = typeFolder === 'syncs' || typeFolder === 'actions';

if (isNested) {
  const providerConfigKey = pathSegments[pathSegments.length - 3];
  providerConfiguration = config.find((config) => config.providerConfigKey === providerConfigKey);
} else {
...
```

Reply:
Thanks, updated